include/gatekeeper_gk.h

```diff
@@ -98,9 +117,17 @@ struct gk_fib {
 
 	union {
 		/*
-	 	 * The nexthop information when the action is GK_FWD_GATEWAY.
+	 	 * The nexthop information when the action is
+		 * GK_FWD_GATEWAY_*_NET.
 	 	 */
-		struct gk_nexthop nexthop;
+		struct {
+			/* The IP address of the nexthop. */
+			struct ipaddr ip_addr;
+
+			/* The cached Ethernet header. */
+			struct ether_cache *eth_cache;
+		} nexthop;
```

Comment:
Renaming this field from 'nexthop' to 'gateway' would improve readability since it'll match the constants `GK_FWD_GATEWAY_*_NET`.